man/rgbasm.1

```diff
@@ -176,6 +177,11 @@ See the
 section for a list of warnings.
 .It Fl w
 Disable all warning output, even when turned into errors.
+.It Fl X Ar max_errors , Fl \-max-errors Ar max_errors
+If more than this number of errors (not warnings) occur, then abort the assembly process.
+.Fl X 0
+disables this behavior.
+The default is 0.
```

Comment:
I think a reasonably low default would make for a better UX than "unlimited"? Something like 100 feels high enough, but perhaps we may want to go lower?

Reply:
Some comparisons:

- C/C++ `gcc`/`clang`: `fmax-errors` defaults to [0](https://gcc.gnu.org/onlinedocs/gcc-4.6.0/gcc/Warning-Options.html) (unlimited)
- Java `javac`: `-xmaxerrs` defaults to [100](https://stackoverflow.com/questions/3115537/java-compilation-errors-limited-to-100)
- Rust: no limit
- Go: the default limit is [10](https://menno.io/posts/showing-all-go-compiler-errors/) (`-gcflags=-e` for all)
- VB.NET: the limit is [100](https://learn.microsoft.com/en-us/dotnet/visual-basic/misc/bc42206)
- Typescript `tsc`: no limit
- nasm: no limit

I think 100 is fine as a default: say each error is 3 lines, that's 300 output lines. Yes it's a lot, but it won't exceed your scrollback buffer, and it also won't be forcing you to do a cycle of "fix, rebuild, read the next batch of errors, repeat".

Reply:
I don't have a particularly strong opinion on this, but I think the UX benefits from limiting the error counts. Possibly we could make this less of an issue by improving "continue-after-error" behaviour, such as not reporting duplicate "X not defined" errors; but this is a longer-term effort, and I think we'd benefit from **setting the max to 100** now.

Let me go on record to say that this default is not fixed in stone, and is subject to be amended later depending on feedback. I don't think we ought to consider it part of our SemVer'd interface.